src/ol_infrastructure/applications/edxapp/__main__.py

```diff
@@ -426,6 +461,246 @@
+        protocol="HTTPS",
+    ),
+    name=edxapp_web_tag[:TARGET_GROUP_NAME_MAX_LENGTH],
+    tags=aws_config.tags,
+)
+edxapp_web_acm_cert = acm.Certificate(
+    "edxapp-load-balancer-acm-certificate",
+    domain_name=edxapp_domains["lms"],
+    subject_alternative_names=[
+        domain for key, domain in edxapp_domains.items() if key != "lms"
+    ],
+    validation_method="DNS",
+    tags=aws_config.tags,
+)
+
+edxapp_acm_cert_validation_records = (
+    edxapp_web_acm_cert.domain_validation_options.apply(
+        partial(acm_certificate_validation_records, zone_id=edxapp_zone_id)
+    )
+)
+
+edxapp_web_acm_validated_cert = acm.CertificateValidation(
+    "wait-for-edxapp-acm-cert-validation",
+    certificate_arn=edxapp_web_acm_cert.arn,
+    validation_record_fqdns=edxapp_acm_cert_validation_records.apply(
+        lambda validation_records: [
+            validation_record.fqdn for validation_record in validation_records
+        ]
+    ),
+)
+edxapp_web_alb_listener = lb.Listener(
+    "edxapp-web-alb-listener",
+    certificate_arn=edxapp_web_acm_validated_cert.certificate_arn,
+    load_balancer_arn=web_lb.arn,
+    port=DEFAULT_HTTPS_PORT,
+    protocol="HTTPS",
+    default_actions=[
+        lb.ListenerDefaultActionArgs(
+            type="forward",
+            target_group_arn=web_lb_target_group.arn,
+        )
+    ],
+)
+
+# Create auto scale group and launch configs for Edxapp web and worker
+cloud_init_user_data = base64.b64encode(
+    "#cloud-config\n{}".format(
+        yaml.dump(
+            {
+                "write_files": [
+                    {
+                        "path": "/etc/consul.d/99-autojoin.json",
+                        "content": json.dumps(
+                            {
+                                "retry_join": [
+                                    "provider=aws tag_key=consul_env "
+                                    f"tag_value={env_name}"
+                                ],
+                                "datacenter": env_name,
+                            }
+                        ),
+                        "owner": "consul:consul",
+                    },
+                ]
+            },
+            sort_keys=True,
+        )
+    ).encode("utf8")
+).decode("utf8")
+
+web_instance_type = edxapp_config.get("web_instance_type") or InstanceTypes.medium.name
+web_launch_config = ec2.LaunchTemplate(
+    "edxapp-web-launch-template",
+    name_prefix=f"edxapp-web-{env_name}-",
+    description=f"Launch template for deploying Edxapp web nodes in {env_name}",
+    iam_instance_profile=ec2.LaunchTemplateIamInstanceProfileArgs(
+        arn=edxapp_instance_profile.arn,
+    ),
+    image_id=edxapp_web_ami.id,
+    vpc_security_group_ids=[
+        edxapp_security_group.id,
+        edxapp_vpc["security_groups"]["web"],
+        consul_security_groups["consul_agent"],
+    ],
+    instance_type=InstanceTypes[web_instance_type].value,
+    key_name="",
```

Comment:
Doesn't this need to be defined?

Reply:
Apparently not since I was able to launch the instances with the code in that state... Definitely should be though.